MoPubSDK/Native Ads/MPCollectionViewAdPlacer.m

```diff
@@ -117,6 +123,19 @@ - (void)adPlacer:(MPStreamAdPlacer *)adPlacer didLoadAdAtIndexPath:(NSIndexPath
 
 - (void)adPlacer:(MPStreamAdPlacer *)adPlacer didRemoveAdsAtIndexPaths:(NSArray *)indexPaths
 {
+    for (const id obj in indexPaths) {
+        if (![obj isKindOfClass:[NSIndexPath class]]) {
+            return;
+        }
+        NSIndexPath *indexPath = obj;
+        NSInteger numberOfItemsInSection = [self.collectionView numberOfItemsInSection:indexPath.section];
+        if (indexPath.section >= self.collectionView.numberOfSections ||
+            indexPath.item >= numberOfItemsInSection) {
+            NSLog(@"mopub-ios-sdk: MPCollectionViewAdPlacer: adPlacer:didRemoveAdsAtIndexPath: IndexPaths: %@ Attempted to delete item %@ in section %@ when number of sections is %@ and number of items in section is %@", indexPaths, @(indexPath.item), @(indexPath.section), @(self.collectionView.numberOfSections), @(numberOfItemsInSection));
+            return;
```

Comment:
same question here re: whether we should exit early or continue with the rest of the array